File: nutrizer/lib/models/user_model.dart
class UserModel {
  String nickname;
  String username;
  String email;
  String birthday;
  double height;
  double weight;
  double bmi;
  String lastLoginTime;
  String lastLoginFrom;
  String token;

  UserModel(
      {this.nickname,
      this.username,
      this.email,
      this.birthday,
      this.height,
      this.weight,
      this.bmi,
      this.lastLoginTime,
      this.lastLoginFrom});

  UserModel.fromJson(Map<String, dynamic> json) {
    nickname = json['nickname'];
    username = json['username'];
    email = json['email'];
    birthday = json['birthday'];
    height = json['height'] != null ? json['height'].toDouble() : 0.0;
    weight = json['weight'] != null ? json['weight'].toDouble() : 0.0;
    bmi = json['bmi'];
    lastLoginTime = json['lastLoginTime'];
    lastLoginFrom = json['lastLoginFrom'];
    token = json['token'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['nickname'] = this.nickname;
    data['username'] = this.username;
    data['email'] = this.email;
    data['birthday'] = this.birthday;
    data['height'] = this.height;
    data['weight'] = this.weight;
    data['bmi'] = this.bmi;
    data['lastLoginTime'] = this.lastLoginTime;
    data['lastLoginFrom'] = this.lastLoginFrom;
    data['token'] = this.token;
    return data;
  }
}

class BmiModel {
  double height;
  double weight;
  double bmi;
  String bmiText;

  BmiModel(
      {
      this.height,
      this.weight,
      this.bmi,
      this.bmiText
      });

  BmiModel.fromJson(Map<String, dynamic> json) {
    height = json['height'] != null ? json['height'].toDouble() : 0.0;
    weight = json['weight'] != null ? json['weight'].toDouble() : 0.0;
    bmi = json['bmi'];
    bmiText = json['bmiText'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['height'] = this.height;
    data['weight'] = this.weight;
    data['bmi'] = this.bmi;
    data['bmiText'] = this.bmiText;
    return data;
  }
}

File: nutrizer_web/application/models/web/User_model.php
<?php

defined('BASEPATH') or exit('No direct script access allowed');

class User_model extends CI_Model
{

    public function insert($data)
    {
        $this->db->insert('tbl_user', $data);
        return $this->db->affected_rows() > 0;
    }

    
    public function update($id, $data)
    {
        $this->db->where('id', $id);
        $this->db->update('tbl_user', $data);
        return $this->db->affected_rows() > 0;
    }

    public function delete($id)
    {
        $this->db->where('id', $id);
        $this->db->delete('tbl_user');
        return $this->db->affected_rows() >0;
    }

    public function getById($id)
    {
        $this->db->select('username,nickname,birthday,gender,email,weight,height,islocked');
        $this->db->where('id', $id);
        $query = $this->db->get('tbl_user');
        $result = $query->row_array();
        if (is_array($result)) {
            return $result;
        } else {
            return false;
        }
    }

    
    public function getListData($limit = 10, $offset = 0,  $search = null)
    {
        $this->db->select("id id,nickname text, nickname,birthday, a.gender");
        $this->db->from("tbl_user a");
        if (!is_null($search)) {
            $this->db->group_start();
            $this->db->like('a.nickname', $search);
            $this->db->or_like('a.username', $search);
            $this->db->group_end();
        }

        $this->db->limit(empty($limit) ? 10 : $limit);
        $this->db->offset(empty($offset) ? 0 : $offset);

        $query = $this->db->get();
        return $query->result();
    }


    public function getListDataTotal($search = null)
    {
        $this->db->select("COUNT(1) total");
        $this->db->from("tbl_user a");

        if (!is_null($search)) {
            $this->db->group_start();
            $this->db->like('a.nickname', $search);
            $this->db->or_like('a.username', $search);
            $this->db->group_end();
        }

        $query = $this->db->get();
        return $query->row_array()['total'];
    }
}
                        
/* End of file user.php */
